core/storage/filestore/impl/filesystem/filesystem_filestore.cpp

```diff
@@ -0,0 +1,43 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "storage/filestore/impl/filesystem/filesystem_filestore.hpp"
+#include "boost/filesystem.hpp"
+#include "storage/filestore/filestore_error.hpp"
+#include "storage/filestore/impl/filesystem/filesystem_file.hpp"
+
+using fc::storage::filestore::File;
+using fc::storage::filestore::FileStoreError;
+using fc::storage::filestore::FileSystemFile;
+using fc::storage::filestore::FileSystemFileStore;
+
+fc::outcome::result<std::shared_ptr<File>> FileSystemFileStore::open(
+    Path &path) noexcept {
+  auto file = std::shared_ptr<File>(new FileSystemFile(path));
+  OUTCOME_TRY(file->open());
+
+  return file;
```

Comment:
will it just rethrow an exception if it happens?

Reply:
No, boost::fstream is derived from std::fstream that has a 2 mechanisms of error handling - error bitmask by default and exceptions that should be turned on (http://www.cplusplus.com/reference/ios/ios/exceptions/). Since I don't turn on exceptions it is not thrown.